src/test/regress/expected/non_colocated_subquery_joins.out

```diff
@@ -0,0 +1,917 @@
+    WHERE
+        foo.user_id = bar.user_id AND
+        foo.user_id IN (SELECT user_id FROM events_table WHERE user_id < 10);
+$$);
+DEBUG:  generating subplan 11_1 for subquery SELECT (users_table.user_id / 2) AS user_id FROM public.users_table, public.events_table WHERE ((users_table.user_id = events_table.user_id) AND (events_table.event_type = ANY (ARRAY[5, 6, 7, 8])))
+DEBUG:  Plan 11 query after replacing subqueries and CTEs: SELECT foo.user_id FROM (SELECT users_table.user_id, events_table.event_type FROM public.users_table, public.events_table WHERE ((users_table.user_id = events_table.user_id) AND (events_table.event_type = ANY (ARRAY[1, 2, 3, 4])))) foo, (SELECT intermediate_result.user_id FROM read_intermediate_result('11_1'::text, 'binary'::citus_copy_format) intermediate_result(user_id integer)) bar WHERE ((foo.user_id = bar.user_id) AND (foo.user_id IN (SELECT events_table.user_id FROM public.events_table WHERE (events_table.user_id < 10))))
+ valid 
+-------
+ t
+(1 row)
+
+-- Subqueries in WHERE and FROM are mixed
+-- In this query, both the joins in the FROM clause is not colocated
+SELECT true AS valid FROM explain_json_2($$
+
+    SELECT
+        foo.user_id
+    FROM
+        (SELECT users_table.user_id, event_type FROM users_table, events_table WHERE users_table.user_id = events_table.user_id AND event_type IN (1,2,3,4)) as foo,
+        (SELECT (users_table.user_id / 2) as user_id FROM users_table, events_table WHERE users_table.user_id = events_table.user_id AND event_type IN (5,6,7,8)) as bar
+    WHERE
+        foo.user_id = bar.user_id AND
+        foo.user_id NOT IN (SELECT user_id FROM events_table WHERE user_id < 10);
+$$);
+DEBUG:  generating subplan 13_1 for subquery SELECT (users_table.user_id / 2) AS user_id FROM public.users_table, public.events_table WHERE ((users_table.user_id = events_table.user_id) AND (events_table.event_type = ANY (ARRAY[5, 6, 7, 8])))
+DEBUG:  generating subplan 13_2 for subquery SELECT user_id FROM public.events_table WHERE (user_id < 10)
+DEBUG:  Plan 13 query after replacing subqueries and CTEs: SELECT foo.user_id FROM (SELECT users_table.user_id, events_table.event_type FROM public.users_table, public.events_table WHERE ((users_table.user_id = events_table.user_id) AND (events_table.event_type = ANY (ARRAY[1, 2, 3, 4])))) foo, (SELECT intermediate_result.user_id FROM read_intermediate_result('13_1'::text, 'binary'::citus_copy_format) intermediate_result(user_id integer)) bar WHERE ((foo.user_id = bar.user_id) AND (NOT (foo.user_id IN (SELECT intermediate_result.user_id FROM read_intermediate_result('13_2'::text, 'binary'::citus_copy_format) intermediate_result(user_id integer)))))
+ valid 
+-------
+ t
+(1 row)
+
+-- Subqueries in WHERE and FROM are mixed
+-- In this query, one of the joins in the FROM clause is not colocated and subquery in WHERE clause is not colocated
+-- similar to the above, but, this time bar is the anchor subquery
+SELECT true AS valid FROM explain_json_2($$
```

Comment:
@marcocitus this might be confusing to you. note that this is another edge we should consider when we implement outer joins via recursive planning. We currently error out for this query because the outer side of the semi-join is intermediate results. However, for this query, the inner side of the join also doesn't contain any distributed tables (e.g., another intermediate result).